overlord/snapstate/autorefresh_test.go

```diff
@@ -50,28 +50,38 @@ type autoRefreshStore struct {
 	ops []string
 
 	err error
+
+	snapActionOpsFunc func()
 }
 
 func (r *autoRefreshStore) SnapAction(ctx context.Context, currentSnaps []*store.CurrentSnap, actions []*store.SnapAction, assertQuery store.AssertionQuery, user *auth.UserState, opts *store.RefreshOptions) ([]store.SnapActionResult, []store.AssertionResult, error) {
-	if assertQuery != nil {
-		panic("no assertion query support")
-	}
-	if !opts.IsAutoRefresh {
-		panic("AutoRefresh snap action did not set IsAutoRefresh flag")
-	}
+	// this is a bit of a hack to simulate race conditions where while the store
+	// has unlocked the global state lock something else could come in and
+	// change the auto-refresh hold
+	if r.snapActionOpsFunc != nil {
+		r.snapActionOpsFunc()
```

Comment:
this should be probably have a return in this case so we avoid the else, and the change is also smaller